datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -128,6 +144,84 @@ impl<T: ArrowPrimitiveType, const NULLABLE: bool> GroupColumn
         }
     }
 
+    fn vectorized_equal_to(
+        &self,
+        group_indices: &[usize],
+        array: &ArrayRef,
+        rows: &[usize],
+        equal_to_results: &mut [bool],
+    ) {
+        let array = array.as_primitive::<T>();
```

Comment:
Here we can check the input arrays to be non-null and optimize for that case

Reply:
I plan to try it together with https://github.com/apache/datafusion/pull/12996#discussion_r1818601807 on follow up prs.
Seems code changes in this one are already a bit large...